### ISSUE TITLE
Internal server error, presumably through flowing water

### ISSUE THREAD

**Issue by Reypatey** [LunaPixelStudios/Better-Minecraft#212]
### Describe the Bug

I've set up a server on my pc with the provided server version for 1.18.2v2 with no additional mods.

I'm joining the server with the corresponding version in my curseforge client. The only additional mod I've installed is Hide Hands 1.18.2-1.5.
Everything runs just fine but when I break a block that held water back I get kicked out of the game saying "Internal Server Error" and no additional feedback. I can't rejoin the server until i edit the nbt value of my character so that i join at a slightly different position. As soon as I move a bit the same issue as before appears, even without manipulating another block.

### Reproduction Steps

1. Dive down a lake
2. Dig the sea bottom to reach a cave
3. Once water starts flowing the bug appears
...

### Screenshots and Videos

https://user-images.githubusercontent.com/102057449/159330063-6f3de898-d5ff-4387-a2c1-fa6ca0f26783.mp4

### Crash Report or Log

https://hidrive.ionos.com/lnk/LsnoWZtN

### Operating System

Windows 10

### Minecraft Version

1.18

### Modloader

Forge

### Modpack Version

bmc v2

### Optifine

No

### Other Mods

Hide Hands hidehands_1.18.2-1.5.jar (Client)

**Comment by SharkieTV** [comment]
Im unable to recreate this, and how do you already have netherite?

**Comment by SharkieTV** [comment]
It says something about Experience Orb, do you have Enchanted gear on you if so can you lmk as it says something about Experience Orb Enchantments.

**Comment by Reypatey** [comment]
I've just now found the time to check again.

I don't think I was right with the crashes being caused by water manipulation but much rather collecting experience orbs, as you suggested.
My gear has several enchantments, all of them we had before in our vanilla world without mods.

I've made myself a backpack from rabbit skin and wanted to try out a few enchantments:
- marksman
- imbued hide
- repairman

I've got the suspicion it could be the repairman enchantment since it's supposed to repair items within your backpack.

To test this out I've made another backpack with just marksman on it and until now I didn't have any more crashes.

Thanks for your help <3
